cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -118,14 +121,17 @@ func (m *asgCache) FindForInstance(instance *azureRef, vmType string) (cloudprov
 	m.mutex.Lock()
 	defer m.mutex.Unlock()
 
+	klog.V(4).Infof("FindForInstance: starts, ref: %s", instance.Name)
 	resourceID, err := convertResourceGroupNameToLower(instance.Name)
+	klog.V(4).Infof("FindForInstance: resourceID: %s", resourceID)
 	if err != nil {
 		return nil, err
 	}
 	inst := azureRef{Name: resourceID}
 	if m.notInRegisteredAsg[inst] {
 		// We already know we don't own this instance. Return early and avoid
 		// additional calls.
+		klog.Warningf("FindForInstance: Couldn't find NodeGroup of instance %q", inst)
```

Comment:
Do we want this to be a warning? This log could be really noisy especially in scenarios where users have multiple node pools but only one with autoscaler enabled. That's going to log one line per virtual machine in the vmss for every non-registered node group.

Reply:
good catch, let's change this to V(4)